### DISCUSSION TITLE
Escrever código Open Source todos os dias?

### DISCUSSION THREAD

**Discussion by afonsopacifer** [frontendbr/forum#163] [Discussão]
Depois que o John Resig (criador do jQuery) fez um desafio pessoal de contribuir com projetos Open Source todos os dias, ele levantou essa questão e acabou motivando outras pessoas a aceitarem esse desafio doido, acabou que o @raphamorim entrou nessa, levando e desafio até 500 dias seguidos de contribuições. Depois que eu assisti uma Talk dele em 2014, acabei entrando nessa, e recentemente conclui um ano de contribuições.

Ao longo desse tempo aprendemos muita coisa e perdemos muito amigos (rsrsr). 

Acabamos gravando um bate papo sobre os lados bons e ruins do desafio:

https://www.youtube.com/watch?v=n9967Iw6Ql0

O que vocês chão desse desafio, não necessariamente todo dia, mas talvez contribuir toda semana? O que vcs acham sobre sair da zona de conforto e se desafiar a ser mais ativo no mundo Open Source?

**Alguns links:**
- [Artigo do John Resig](http://ejohn.org/blog/write-code-every-day/)
- [Palestra do Rapha na BrazilJS](https://www.youtube.com/watch?v=toCdZ2e9Dh4)
- [Projeto que criamos para motivar a galera](http://writecodeeveryday.io/)

**Comment by kazzkiq** [comment]
Eu acho muito bacana!

Tudo que eu sei hoje eu devo a comunidade open-source. Tutoriais gratuitos, projetos abertos que sempre me salvavam no último minuto... é um mundo livre, grátis, e belo! E nunca daria certo sem esse apoio tão grande da comunidade.

Eu particularmente gostaria de ajudar muito mais. Mas falta tempo (eterno inimigo dos devs) e muitas vezes eu tenho medo de assumir um compromisso e depois deixar a galera na mão. Por conta disso, eu prefiro mudar a estratégia: Sempre que trabalho em um projeto, eu tento selecionar as partes dele que eu ache que possam ser úteis para outras pessoas e solto em forma de mini-projetinhos open-source. É vantajoso pra mim (já que eu teria que fazer eles de qualquer forma para os meus projetos), é vantajoso para a comunidade e é uma forma de retribuir tudo que o mundo open-source já me deu de bom nesse período todo. Até agora já consegui lançar dois projetos nesse modelo, e tem mais dois no forno. Todos eles foram úteis pra mim e hoje são úteis para a comunidade (espero eu 😄 ).

Numa parte do seu vídeo vocês falam sobre haterismo (e é incrível que isso exista até no meio open-source). Mas eu acho que a melhor resposta pra esse tipo de coisa é a indiferença. Você ta ali contribuindo pra comunidade e ajudando outras pessoas e o cara vem falar mal? Deixa falando com a parede! Hahaha

**Comment by wellingtonpgp** [comment]
Eu acho muito bacana! 
Apenas falta um pouco te tempo para escrever, mas apoio totalmente quem escreve código open source, isso ajuda muito a você a praticar mais e a comunidade de devs em geral,  o lado ruim disso e que tem pessoas que cria projetos legais, a galera começa a usar mas o criador não da continuidade no código, já tive esse problemas com diverso códigos  open.

**Comment by Marcelosilva10** [comment]
Ótimo vou abraçar essa causa.

**Comment by felipefialho** [comment]
Fala @afonsopacifer. 

Assisti o vídeo na sexta-feira, algumas questões:

#### Pontos positivos
- A ideia é realmente legal, você estabelece uma meta e se compromete em cumprir. 
- De quebra você melhora diversas habilidades nesse período e ajuda (ou lança) diversos projetos que você usa no dia a dia. 
- É quase como devolver um pouco do que recebeu. 

#### Pontos negativos
- Sinceramente, acho surreal utilizar dos fins de semana / feriados para esse fim, vi o @raphamorim comentar que levava notebook em viagens para cumprir a meta
- Como vocês disseram... você até lança projetos, o problema é a manutenção. E concordo totalmente! Tenho alguns projetos lançados e é extremamente difícil deixa-los atualizados ou fazer melhorias. 
- É clichê, mas tudo em excesso faz mal. E se comprometer com tal meta, praticamente te obriga a deixar de lado sua vida pessoal, sem equilíbrio a tendencia é que as coisas saiam do controle com o passar do tempo.

---

Sobre mim:

Tenho mantido um [Daylog](https://github.com/LFeh/dailylog) para documentar as coisas que ando lendo e para me forçar a ler um pouco todo dia. Sendo assim, faço ao menos um commit por dia de segunda a sexta.

Porém tenho me tentado me policiar com relação a projetos open source. Tenho tentado separar cerca de 1h, três vezes por semana (sem atrapalhar outras tarefas da minha vida). Nessa pegada:
- Desenvolvi a parte de interface do [Eventos](https://github.com/frontendbr/eventos/tree/interface)
- [Escrevi um artigo no Medium](https://medium.com/@lfeh/why-every-developer-must-to-go-to-events-6a5327a977f9#.hdj4xccmx) 
- Criei o [Kratos](https://github.com/LFeh/kratos-boilerplate) 
- Atualizei meu [Coding Style](https://github.com/LFeh/coding-style)

Sem sair dos trilhos e sem ficar maluco. Agora estou tentando atualizar meu site e terminando essa tarefa pulo para outra coisa e assim vai.

Além dos projetos Open Source também cuido do [Meetup CSS](www.meetup.com/pt-BR/CSS-SP/) junto com o @raphaelfabeni e demanda bastante energia da gente.

Nesses anos como desenvolvedor aprendi que por mais que nossa profissão seja desafiadora, precisamos muito buscar equilíbrio, ou corremos o risco de ter problemas como reclusão social, ansiedade, etc...

**Comment by yanmagale** [comment]
E ai @afonsopacifer . Assim como o @LFeh eu também assisti o vídeo e acho muito incrível a iniciativa. Fico imaginando a enorme quantidade de benefícios e conhecimento que você adquire com esta prática.

Contudo, concordo com a fala do @LFeh sobre o aspecto de ter certos limites e qualidades de vida. Uma questão que cabe também ser avalidada é o momento de vida de cada um, para saber como encaixar esta prática em seu dia-a-dia. Eu por exemplo, trabalho o dia todo, faço faculdade a noite, ajudo a organizar comunidade de desenvolvedores (GDGBH) e ainda tenho que estudar todo esse universo de tecnologias da web. Assim como eu, também existem pessoas que por N motivos, não podem aderir a prática diária. 

Acho que cabe a cada um avaliar o seu momento, mensurar o seu tempo e ir incluindo esta prática em seu cotidiano. Inclusive vou até iniciar o meu Daily Log (o meu ainda é nos bloquinhos de papel) e incluir práticas de contruibuição, pois sei que é uma prática que ajuda e muito na evolução e também poder contribuir em projetos abertos.

**Comment by raphamorim** [comment]
Concordo com todos.

Mas a ideia do Write Code Everyday é o mesmo `feel` do Daily Log não? Para atualizar seu log você precisa commitar algo usando `git` em um repo aberto. O Write Code Everyday surgiu com a mesma ideia. Você não precisa escrever projetos abertos por semana.

Crie um repo de coisas que vem fazendo ou aprendendo, de certa forma você vai aprender coisas novas até mesmo usando GH todo dia.

Yan, mas tempo é algo que cada pessoa mensura de forma diferente. Eu fazia faculdade a noite, trabalhava, levava 4 horas para chegar em casa (ida e volta) e organizava o riojs, assim como você. Mas ainda sim conseguia tirar 20min para escrever algum código que fosse. 

Tinha dias que era produtivo, outros não. O Jaydson explica muito bem isso em um [artigo](http://jaydson.org/a-falacia-do-nao-ter-tempo/).

#### Sobre o ponto do **overload** do @LFeh

Cara super concordo e foi o que me fez parar, teria parado bem antes até. Queria viajar, sair com os amigos e tudo mais, mas estava preso. Nesse ponto, acho que atrapalha.

Mas contudo, eu nunca teria contribuido com features do core de repos do jQuery e Mozilla, não tinha disciplina para abrir issues, estudar e correr atrás de implementar. Não teria aprendido tanto em pouco tempo. Então eu acho que sim, o desafio é válido.

Mas dois pesos, duas medidas. **Não recomendo escrever código todo dia**, acho válido tirar 3-4 dias da semana para executar o desafio, afinal **viver é muito mais importante** (coisa que venho fazendo hoje e ta dando super certo).

**Comment by lagden** [comment]
Eu escrevo todos os dias...
Vai fluindo naturalmente...

O workflow é assim:
- Tenho uma necessidade ou idéia
- Então procuro algo que necessito e tento ajudar...
- Se não eu escrevo o meu próprio...

Repos que dei um helps:
https://github.com/pulls?page=1&q=is%3Apr+author%3Alagden&utf8=%E2%9C%93

Libs pra ajudar no dia a dia:
https://github.com/lagden/form-validation
https://github.com/lagden/promisify
https://github.com/lagden/tooltip
https://github.com/lagden/utils
https://github.com/lagden/postcss-inline-base64
https://github.com/lagden/io-cep
https://github.com/lagden/svg-localstorage
https://github.com/lagden/growl

E assim segue a vida!!

**Comment by yanmagale** [comment]
:+1:  @raphamorim , concordo 100% com você cara.
